etl/grapher_helpers.py

```diff
@@ -644,16 +645,24 @@ def add_columns_for_multiindicator_chart(
     """
     table = table.copy()
 
-    # Fill columns with zeros.
+    def _rename_column(old_column_name: str, chart_slug: str) -> str:
+        # Generate a name for a new column based on the old column name and the chart slug.
+        return f"{old_column_name}_chart_{underscore(chart_slug)}"
+
+    # Create new columns.
+    new_columns = [_rename_column(old_column_name=column, chart_slug=chart_slug) for column in columns_in_chart]
+    table[new_columns] = table[columns_in_chart].copy()
+
+    # Optionally fill some of the new columns with zeros.
     if columns_to_fill_with_zeros is not None:
         # Sanity check.
         error = "columns_to_fill_with_zeros should be a subset of columns_in_chart."
         assert set(columns_to_fill_with_zeros) <= set(columns_in_chart), error
-        table[columns_to_fill_with_zeros] = table[columns_to_fill_with_zeros].fillna(0)
-
-    # Create new columns.
-    new_columns = [f"{column}_chart_{underscore(chart_slug)}" for column in columns_in_chart]
-    table[new_columns] = table[columns_in_chart].copy()
+        # Fill nans with zeros (in new columns).
+        new_columns_to_fill_with_zeros = [
+            _rename_column(old_column_name=column, chart_slug=chart_slug) for column in columns_to_fill_with_zeros
+        ]
+        table[new_columns_to_fill_with_zeros] = table[new_columns_to_fill_with_zeros].fillna(0)
```

Comment:
I'm not sure it is working as expected, variables that are in `columns_to_fill_with_zeros`, get their zeros here and then these zeros are then replaced with NaNs with `table.loc[table[new_columns].isnull().any(axis=1), new_columns] = np.nan`. I think the zero-filled variables need a different column name.

Reply:
Yes, this is the behaviour I needed:
* I first fill with zeros those variables that are not too important and often missing (e.g. "Other renewables"), to avoid having an empty chart where important variables (e.g. "Gas") were available but hidden because of the nans in less important variables.
* Then, if any variable is missing, then all variables should be missing in the same row. That way we ensure that *either we show data for all indicators on a specific country-year, or we show no data*. I think that's the key ingredient to avoid having strange jumps, or to have years where a country has, e.g. 100% renewables because fossil fuels were not informed.

If you simply wanted to fill some indicators with zeros, regardless of the other indicators, I suppose you wouldn't need this function. But if you think that could be the desired outcome in some circumstances, I could add an option to the function, so that the nans can be imposed only in `set(new_columns) - set(new_columns_to_fill_with_zeros)`.

Reply:
If filling with zeros is all you need for a certain chart, you could simply do fillna(0) and rename those indicators as "*_zero_filled", as we used to do in the past.
Alternatively, you could simply do nothing at all, because grapher will by default treat nans as zeros.

Reply:
Ah, I see, I misunderstood but am with you now!

I am having some issues running it locally and am currently rebuilding my staging server so currently can't check out the output data just yet. However, feel free to merge so the electricity mix issues will get solved and I can double-check data once I am able.